staging/src/k8s.io/client-go/scale/interfaces.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package scale
+
+import (
+	autoscalingapi "k8s.io/api/autoscaling/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ScalesGetter can produce a ScaleInterface
+// for a particular namespace.
+type ScalesGetter interface {
+	Scales(namespace string) ScaleInterface
+}
+
+// ScaleInterface can fetch and update scales for
+// resources in a particular namespace which implement
+// the scale subresource.
+type ScaleInterface interface {
+	// Get fetches the scale of the given scalable resource.
+	Get(kind schema.GroupKind, name string) (*autoscalingapi.Scale, error)
+
+	// Update updates the scale of the the given scalable resource.
+	Update(ind schema.GroupKind, scale *autoscalingapi.Scale) (*autoscalingapi.Scale, error)
```

Comment:
Why aren't you using the hub you defined?  If you want to use `autoscaling/v1` as your hub, why define a new internal to convert to?

Reply:
I considered it, but I figured we didn't want to ever expose internal versions in clients.  It does make some sense to me to use it, though (although it also make some sense to use internal versions elsewhere, so /me shrugs).

On the flip side, even though we don't expose it, having an internal version makes it a bit easier to reason about the conversions (IMO), especially since it's following the normal Kubernetes hub-and-spoke pattern.  AFAIK, there's not a technical reason why we couldn't use external as the hub, though.

Reply:
> On the flip side, even though we don't expose it, having an internal version makes it a bit easier to reason about the conversions (IMO), especially since it's following the normal Kubernetes hub-and-spoke pattern. AFAIK, there's not a technical reason why we couldn't use external as the hub, though.

I think this demonstrates the inherent problem of not exposing internal types.  If you want to keep your own hub, I can live with it.  You need fuzzers though.